File: ExerciciosEmJava/faculdade_1semestre/Nivel6_funcoes/Ex05.java
// /Escreva um programa que tenha uma função menu() para exibir
// opções e outras funções para executar as ações de cada opção.

package faculdade_1semestre.nivel6_funcoes;
import java.util.Scanner;

public class Ex05
{
    static Scanner sc = new Scanner(System.in);
    public static void main(String[] args) throws InterruptedException
    {
        menu();
    }

    static void menu() throws InterruptedException
    {
        boolean isOn = true;
        do
        {
            System.out.print("""
            Bem vindo ao menu de funçoes, abaixo segue uma lista com as funçoes, escolha uma de sua preferencia

            1- Hello World
            2- Print numeros
            3- Exit

            Opção: """);
            
            switch (sc.nextInt())
            {
                case 1:
                    System.out.println("Hello World");
                    Pausar();
                    break;
                case 2:
                    int limite = 0;
                    System.out.print("Digite um numero, e contarei todos os numeros do 1 ate o seu numero\nNumero: ");
                    limite = sc.nextInt();
                    PrintNumeros(limite);
                    Pausar();
                    break;
                case 3:
                    System.out.println("Encerrando o menu, muito obrigado por usar os serviços MrNET");
                    Loading();
                    isOn = false;
                    break;
                default:
                    System.out.println("Invalido, por favor, insira novamente. Retornando ao menu principal");
                    Loading();
                    break;
            } 
        } while (isOn);
    }

    private static void PrintNumeros (int num)
    {
        for (int i = 1; i <= num; i++)
        {
            System.out.println(i);
        }
    }

    private static void Loading () throws InterruptedException
    {
        for (var i = 0; i < 3; i++)
        {
            System.out.print(".");
            Thread.sleep(1000);
        }
    }

    private static void Pausar ()
    {
        System.out.println("\nAperte qualquer tecla para retornar ao menu.");
        sc.next();
    }
    
}

File: ExerciciosEmCSharp/Ex05.cs
// /Escreva um programa que tenha uma função menu() para exibir
// opções e outras funções para executar as ações de cada opção.


Ex05.menu();


public static class Ex05
{
    public static void menu ()
    {
        bool isOn = true;
        do
        {
            Console.Clear();
            Console.Write("""
            Bem vindo ao menu de funçoes, abaixo segue uma lista com as funçoes, escolha uma de sua preferencia

            1- Hello World
            2- Print numeros
            3- Exit

            Opção: 
            """);

            switch (int.Parse(Console.ReadLine()))
            {
                case 1:
                    Console.Clear();
                    Console.WriteLine("Hello World");
                    Pausar();
                    break;
                case 2:
                    Console.Clear();
                    int limite = 0;
                    Console.Write("Digite um numero, e contarei todos os numeros do 1 ate o seu numero\nNumero: ");
                    limite = int.Parse(Console.ReadLine());
                    PrintNumeros(limite);
                    Pausar();
                    break;
                case 3:
                    Console.Clear();
                    Console.WriteLine("Encerrando o menu, muito obrigado por usar os serviços MrNET");
                    Loading();
                    
                    isOn = false;
                    break;
                default:
                    Console.WriteLine("Invalido, por favor, insira novamente. Retornando ao menu principal");
                    Loading();
                    break;
            } 
        } while (isOn);
    }

    private static void PrintNumeros (int num)
    {
        for (int i = 1; i <= num; i++)
        {
            Console.WriteLine(i);
        }
    }

    private static void Loading ()
    {
        Console.Write(".");
        Thread.Sleep(1000);
        Console.Write(".");
        Thread.Sleep(1000);
        Console.Write(".");
        Thread.Sleep(1000);
    }

    private static void Pausar ()
    {
        Console.WriteLine("Aperte qualquer tecla para retornar ao menu.");
        Console.ReadKey();
    }
}